Documentation/architecture.md

```diff
@@ -0,0 +1,58 @@
+# Architecture
```

Comment:
Please link to this file from the README.

Reply:
done as well